scripts/prepdocslib/textsplitter.py

```diff
@@ -42,59 +42,63 @@ def find_page(offset):
 
         all_text = "".join(page.text for page in pages)
         length = len(all_text)
-        start = 0
-        end = length
-        while start + self.section_overlap < length:
-            last_word = -1
-            end = start + self.max_section_length
-
-            if end > length:
-                end = length
-            else:
-                # Try to find the end of the sentence
+
+        if length <= self.max_section_length:
```

Comment:
I see - the diff is coming from the indentation after the else?

Reply:
Hi @mattgotteiner ,

I've added if statement befor while loop. For small documents where amount of text is less than section_overlap the condition: 'while start + self.section_overlap < length:' is always false same as the last if statement in the function ('if start + self.section_overlap < end:') so no section is return. For this solution, the assumptions is that section_overlap < max_section_length - but it seems to be obvious.
I've also reported bug related to usage of section_overlap in while loop: https://github.com/Azure-Samples/azure-search-openai-demo/issues/1153, but this is separate case.
Hope, this make it more clear.

Best regards,
Marek

Reply:
Thanks, this makes sense to me, I will manually test this and add some test cases. Appreciate it!